contracts/ts/genMaciState.ts

```diff
@@ -253,36 +277,66 @@ const genMaciStateFromContract = async (
-        ...pollContract.filters.MergeMaciStateAqSubRoots(),
-        fromBlock: fromBlock,
-    })
-
-    const mergeMaciStateAqLogs = await provider.getLogs({
-        ...pollContract.filters.MergeMaciStateAq(),
-        fromBlock: fromBlock,
-    })
-
-    const mergeMessageAqSubRootsLogs = await provider.getLogs({
-        ...pollContract.filters.MergeMessageAqSubRoots(),
-        fromBlock: fromBlock,
-    })
+    // fetch poll contract logs
+    let publishMessageLogs: any[] = []
+    let topupLogs: any[] = []
+    let mergeMaciStateAqSubRootsLogs: any[] = []
+    let mergeMaciStateAqLogs: any[] = []
+    let mergeMessageAqSubRootsLogs: any[] = []
+    let mergeMessageAqLogs: any[] = []
+
+    for (let i = fromBlock; i < lastBlock; i += blocksPerRequest + 1) {
+        const toBlock = (i + blocksPerRequest) >= lastBlock ? undefined : i + blocksPerRequest
+
+        const tmpPublishMessageLogs = await provider.getLogs({
+            ...pollContract.filters.PublishMessage(),
+            fromBlock: i,
+            toBlock
+        })
+        publishMessageLogs = publishMessageLogs.concat(tmpPublishMessageLogs)
+    
+        const tmpTopupLogs = await provider.getLogs({
+            ...pollContract.filters.TopupMessage(),
+            fromBlock: i,
+            toBlock,
+            address: pollContract.address
+        })
+        topupLogs = topupLogs.concat(tmpTopupLogs)    
+    
+        const tmpMergeMaciStateAqSubRootsLogs = await provider.getLogs({
+            ...pollContract.filters.MergeMaciStateAqSubRoots(),
+            fromBlock: i,
+            toBlock,
+            address: pollContract.address
+        })
+        mergeMaciStateAqSubRootsLogs = mergeMaciStateAqSubRootsLogs.concat(tmpMergeMaciStateAqSubRootsLogs)
+    
+        const tmpMergeMaciStateAqLogs = await provider.getLogs({
+            ...pollContract.filters.MergeMaciStateAq(),
+            fromBlock: i,
+            toBlock,
+            address: pollContract.address
+        })
+        mergeMaciStateAqLogs = mergeMaciStateAqLogs.concat(tmpMergeMaciStateAqLogs)
+    
+        const tmpMergeMessageAqSubRootsLogs = await provider.getLogs({
+            ...pollContract.filters.MergeMessageAqSubRoots(),
+            fromBlock: i,
+            toBlock,
+            address: pollContract.address
+        })
+        mergeMessageAqSubRootsLogs = mergeMessageAqSubRootsLogs.concat(tmpMergeMessageAqSubRootsLogs)
+    
+        const tmpMergeMessageAqLogs = await provider.getLogs({
+            ...pollContract.filters.MergeMessageAq(),
+            fromBlock: i,
+            toBlock,
+            address: pollContract.address
+        })
+        mergeMessageAqLogs = mergeMessageAqLogs.concat(tmpMergeMessageAqLogs)
 
-    const mergeMessageAqLogs = await provider.getLogs({
-        ...pollContract.filters.MergeMessageAq(),
-        fromBlock: fromBlock,
-    })
+        if (sleepAmount) await sleep(sleepAmount)
```

Comment:
```suggestion
        if (sleepAmount) {
            await sleep(sleepAmount)
        }
```

nit: while concise, one-liners like this can compromise code readability and may lead to overlooked errors during future edits. It's generally a good practice to expand these into full if statements with braces for clarity.